sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java

```diff
@@ -116,17 +119,38 @@ private RntbdServiceEndpoint(
         this.requestTimer = timer;
 
         this.tag = Tag.of(TAG_NAME, RntbdMetrics.escape(this.remoteAddress.toString()));
+        this.clientMetricTag = Tag.of(
+            TagName.ServiceEndpoint.toString(),
+            String.format("%s_%d", this.serverKey.getHost(), this.serverKey.getPort()));
         this.id = instanceCount.incrementAndGet();
         this.provider = provider;
 
-        this.metrics = new RntbdMetrics(provider.transportClient, this);
         this.maxConcurrentRequests = config.maxConcurrentRequestsPerEndpoint();
 
         this.connectionStateListener = this.provider.addressResolver != null && config.isConnectionEndpointRediscoveryEnabled()
             ? new RntbdConnectionStateListener(this) : null;
 
         this.channelPool = new RntbdClientChannelPool(this, bootstrap, config, clientTelemetry, this.connectionStateListener);
-        this.clientTelemetry = clientTelemetry;
+
+        if (clientTelemetry != null &&
+            clientTelemetry.isClientMetricsEnabled()) {
+
+            RntbdMetricsCompletionRecorder rntbdMetricsV2 =
+                ClientTelemetryMetrics.createRntbdMetrics(provider.transportClient, this);
+            if (RntbdMetrics.isEmpty()) {
+                this.metricsComplectionRecorder = rntbdMetricsV2;
+            } else {
+                List<RntbdMetricsCompletionRecorder> metricCompletionRecorders = new ArrayList<>();
+                metricCompletionRecorders.add(RntbdMetrics.create(provider.transportClient, this));
+                metricCompletionRecorders.add(rntbdMetricsV2);
+                this.metricsComplectionRecorder = new RntbdMetricsDelegatingCompletionRecorder(
+                    metricCompletionRecorders
+                );
+            }
+
+        } else {
+            this.metricsComplectionRecorder = RntbdMetrics.create(provider.transportClient, this);
```

Comment:
should we check RntbdMetrics.isEmpty() here?

Reply:
Not necessary - is isClientMetricsEnabled is false we know only the old legacy metrics would need to be collected (teh ones David initially added) - which is what the RntbdMetrics.create would do. isEmpty only needs ot be checked when deciding whether we just need to emit the new metrics or both.